forge-game/src/main/java/forge/game/trigger/TriggerSpellAbilityCastOrCopy.java

```diff
@@ -248,6 +248,22 @@ public final boolean performTest(final Map<AbilityKey, Object> runParams) {
             }
         }
 
+        if (hasParam("ManaFrom")) {
+            boolean found = false;
+            for (Mana m : spellAbility.getPayingMana()) {
+                Card source = m.getSourceCard();
+                if (source != null) {
+                    if (source.isValid(getParam("ManaFrom"), source.getController(), source, spellAbility)) {
```

Comment:
shouldn't that be matchValidParam ? and shouldn't it be using the trigger as ctb ?